packages/node-matter/package.json

```diff
@@ -0,0 +1,42 @@
+{
+    "name": "@project-chip/node-matter.js",
+    "version": "0.0.1",
+    "description": "Matter protocol for node.js",
+    "keywords": [
+        "iot",
+        "home automation",
+        "matter",
+        "smart device"
+    ],
+    "license": "Apache-2.0",
+    "author": "Project CHIP Authors",
+    "contributors": [
+        "Martin Turon <mturon@google.com>",
+        "Marco Fucci di Napoli <mfucci@gmail.com>"
+    ],
+    "bugs": {
+        "url": "https://github.com/project-chip/matter.js/issues"
+    },
+    "homepage": "https://github.com/project-chip/matter.js",
+    "repository": {
+        "type": "git",
+        "url": "https://github.com/project-chip/matter.js.git"
+    },
+    "scripts": {
+        "clean": "rm -rf dist",
+        "build": "npm run clean; tsc",
+        "test": "TZ=utc true || mocha"
+    },
+    "devDependencies": {
+        "ts-node": "^10.9.1",
+        "typescript": "^4.9.5"
+    },
+    "dependencies": {
+        "@project-chip/matter.js": "^0.1.0"
```

Comment:
How does typescript knows to link against the local matter.js build vs the published one?

Reply:
~~magic~~ [npm workspaces](https://docs.npmjs.com/cli/v9/using-npm/workspaces)

TL;DR: When a dependency matches the package name of a package referenced by the top-level `package.json` `workspaces`, it automatically gets symlinked instead of installed from elsewhere.

TypeScript just works, it doesn't care where the package comes from btw.

Reply:
If you want to try yourself:
```jsonc
// file: package.json
{
	"name": "monorepo",
	"private": true,
	"workspaces": [
		"packages/*"
	]
}
```

```jsonc
// file: packages/a/package.json
{
	"name": "package-a",
	"version": "1.0.0"
}
```

```jsonc
// file: packages/b/package.json
{
	"name": "package-b",
	"version": "1.0.0",
	"dependencies": {
		"package-a": "1.0.0"
	}
}
```

run `npm install` and see how the folders in `node_modules` are actually symlinks to the `packages` subdirectories.

Reply:
Exxactly. This is the magic of the workspaces.

Reply:
![Bildschirm­foto 2023-02-24 um 17 58 18](https://user-images.githubusercontent.com/11976694/221240518-29e6e4c3-35b3-4a71-9ae2-bf72c783a70a.png)

Here it how it looks in the FS in exactly our case